docs/linux/setup_ubuntu-host_android-arm32.md

```diff
@@ -0,0 +1,123 @@
+# Setup: Ubuntu host, arm32 kernel on an Android device
+
+This document will detail the steps involved in setting up a Syzkaller instance fuzzing an ARM32 linux kernel on an Android (or Android Things) device. This is a work-in-progress at this time and being provided to spur further development. Some features of syzkaller may not yet work on ARM32. For example, not all debugging and test coverage features are available in the Linux kernel for ARM32, limiting the efficacy of syskaller in finding bugs fast. These instructions help set up syzkaller to be a basic fuzzer that does not rely on test coverage data from the kernel. 
+
+## Install Android and Linux kernel on an ARM32 device
+
+Follow the instructions for the ARM32 board to install Android or 
+Android Things and make sure the device boots properly.
+
+Set up the adb bridge so that adb and fastboot work.
+
+Setup a serial port, following the instructions for your board so that you can monitor any messages from the kernel.
+
+These were tested on an NXP Pico-Pi-IMX7D following the instructions at:
+
+```
+
+https://developer.android.com/things/hardware/developer-kits.html.
+
+```
+
+
+If feasible, recompile and reinstall the Linux kernel with any debugging options available on your board.
+
+
+## Optional: Setup the NDK cross-compiler toolchain
+
+
+You can find the NDK packages here:
+
+https://developer.android.com/ndk/downloads/index.html
+
+
+Download the package for your host system (linux-x86_64). After you have unzipped it, you should have a directory named android-ndk-r15c, or the latest stable release. We will refer to this directory as $(NDK).
+
+
+Generate a standalone toolchain for the target device in a destination of your choice (assumed to be defined in the variable NDKARM). The destination folder $NDKARM will get created by the code below.
+
+```
+export NDKARM=~/armtc
+cd $NDK
+./build/tools/make_standalone_toolchain.py --arch arm --api 26 --stl=libc++ --install-dir $NDKARM
+```
+
+## Install Go
+
+Install Go:
+``` bash
+wget https://storage.googleapis.com/golang/go1.8.1.linux-amd64.tar.gz
```

Comment:
Let's use go1.9.2 as the latest Go version. It also doesn't require setting GOPATH, so these steps will go away.